### PR TITLE
bugfix: fix code compiling failed with rust-1.49.0

### PR DESCRIPTION
**Pull request by Apokleos**
rust v1.49.0 has stricter constraint, causing Nydus code
compiling failed. This PR will sovle it.

Fixes: #39

Signed-off-by: LiYa'nan <oliverliyn@gmail.com>

### CODE REVIEW

**Comment by bergwolf** [comment]
While at it, please change `.travis.yaml` to use rustc v1.49.0 so that we don't introduce breakage accidentally.

**Comment by Apokleos** [comment]
> While at it, please change `.travis.yaml` to use rustc v1.49.0 so that we don't introduce breakage accidentally.

Thank you @bergwolf 
I have updated the rust version to 1.49.0, but CI is still failed. 
Cloud you give some advice?

**Comment by bergwolf** [comment]
@Apokleos Sure, your change looks good. Let me take a closer look.

File: src/bin/nydus-image/builder.rs [hunk 1]
```diff
@@ -475,10 +478,12 @@ impl Builder {
         let (blob_hash, blob_size, mut blob_readahead_size) = self.dump_blob()?;
 
         // Set blob hash as blob id if not specified.
-        if self.blob_id == "" {
+        if !self.blob_id.is_empty() {
```
[the remark this answers is not in the document]
Comment (Apokleos) [reply src/bin/nydus-image/builder.rs]: Sorry for that， I have corrected it.

**Comment by bergwolf** [comment]
Turns out we do have multiple places to specific rust version for different tools. Please also change the smoke Dockerfile as well... Thanks a lot!

https://github.com/dragonflyoss/image-service/blob/master/misc/smoke/Dockerfile#L1

**Review by bergwolf — APPROVED** [review]
Thanks a lot!

**Pull request merged**